test/integration/glue_test.py

```diff
@@ -1,121 +1,280 @@
+    ids_right: str,
+    id_cols: list[str],
+    save_unmapped_ids,
+    test_dir,
+    request,
+):
+    """Test that the stored ids are internally consistent.
+
+    Args:
+        ids_left: name of fixure cooresponding to a dataframe which contains ID's
+        ids_right: name of fixure cooresponding to a dataframe which contains ID's
+        id_cols: list of ID column(s)
+
+    Raises:
+        AssertionError:
+    """
+    missing = get_missing_ids(
+        request.getfixturevalue(ids_left),
+        request.getfixturevalue(ids_right),
+        id_cols,
     )
-    # Find the indices of all utilities with unmapped plants that do not appear
-    # in the list of mapped utilities at all:
-    new_utilities_index = utils_with_unmapped_plants.index.difference(
-        mapped_utilities.index
+    if save_unmapped_ids:
+        save_to_devtools_glue(df=missing, test_dir=test_dir, request=request)
+    if not missing.empty:
+        raise AssertionError(f"Found {len(missing)} {id_cols}: {missing}")
+
+
+@pytest.mark.parametrize(
+    "ids_left,ids_right,id_cols,drop",
+    [
+        pytest.param(
+            "plants_ferc1",
+            "plants_ferc1_raw",
+            ["utility_id_ferc1", "plant_name_ferc1"],
+            (227, "comanche"),
+            id="check_for_unmmapped_plants_in_plants_ferc1",
+        ),
+        pytest.param(
+            "utilities_ferc1",
+            "utilities_ferc1_xbrl",
+            ["utility_id_ferc1"],
+            (227),
+            id="validate_utility_id_ferc1_in_utilities_ferc1_xbrl",
+        ),
+    ],
+)
+def test_for_unmapped_ids_minus_one(
+    ids_left: str, ids_right: str, id_cols: list[str], drop: tuple, request
+):
+    """Test that we will find one unmapped ID after dropping one.
+
+    Args:
+        ids_left: name of fixure cooresponding to a dataframe which contains ID's
+        ids_right: name of fixure cooresponding to a dataframe which contains ID's
+        id_cols: list of ID column(s)
+        drop: a tuple of the one record IDs to drop
+
+    Raises:
+        AssertionError:
+    """
+    ids_minus_one = (
+        request.getfixturevalue(ids_left).set_index(id_cols).drop(drop).reset_index()
     )
-    # Use that index to select only the previously unmapped utilities:
-    unmapped_utilities = utils_with_unmapped_plants.loc[
-        new_utilities_index
-    ].reset_index()
-    if len(unmapped_utilities) != 1:
-        raise AssertionError(
-            f"Found {len(unmapped_utilities)} "
-            f"unmapped FERC 1 utilities instead of 1."
-        )
+    missing = get_missing_ids(
+        ids_minus_one, request.getfixturevalue(ids_right), id_cols
+    )
+    if len(missing) != 1:
+        raise AssertionError(f"Found {len(missing)} {id_cols} but expected 1.")
 
 
-def test_unmapped_plants_eia(pudl_settings_fixture, pudl_engine):
-    """Check for unmapped EIA Plants."""
-    unmapped_plants_eia = get_unmapped_plants_eia(pudl_engine)
+def test_unmapped_plants_eia(
```

Comment:
my current intention is to separate the "find the missing id's" and the "label the missing id's for manual mapping" sets. This will enable me to integrated these two EIA tests into `test_for_fk_validation_and_unmapped_ids`